ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -200,10 +200,65 @@ function NetworkGraphPage() {
                     // get active nodes from network flow graph API response
                     const { nodes: activeNodes } = values[0].response;
+
+                    /*
+                    FIX for 4.3.x central <-> 4.4.x sensor mismatch that causes the Network Graph to not render
+
+                    This is a patch to fix a compatibility issue between central and sensor, where sensor sends an entity
+                    type of `5`, which is not understood by central and passed to the UI directly as a numeric value
+                    instead of being converted to a string enum of `INTERNAL_ENTITIES` as it is in central >= 4.4.0
+
+                    The fix is to convert all of these numeric value types to the external entity type of `INTERNET` immediately when
+                    the graph responses are received, which will be equivalent to the behavior pre-4.4.
+
+                    To do so, we iterate through both the policy and active nodes:
+
+                        If there is an INTERNET entity type in the nodes, save it, as we need to use this as the entity for all
+                            nodes with an invalid entity type. Otherwise, there will be two instances of the "External Entities"
+                            node displayed in the graph.
+
+                        Any nodes with an entity type of `5` or `INTERNAL_ENTITIES` are invalid and need to be added to a list
+                            to be converted to the INTERNET entity type.
+
+                    Then iterate through the invalid entity nodes:
+
+                        If there is an existing INTERNET entity type in the nodes, set the `entity`
+                            field of each invalid node to the saved INTERNET entity.
+
+                        If there is no INTERNET entity type in the nodes, we convert the entity type to `INTERNET`
+                            directly, as all entities with type `5` will share the same ID.
+                   */
+                    let existingInternetEntityNode: any;
+                    const invalidEntityNodes: any[] = [];
+                    [...policyNodes, ...activeNodes].forEach((node) => {
+                        if (node?.entity?.type === 'INTERNET') {
+                            existingInternetEntityNode = node;
+                        }
+                        if (
+                            node?.entity?.type === 5 ||
+                            node?.entity?.type === 'INTERNAL_ENTITIES'
+                        ) {
+                            invalidEntityNodes.push(node);
+                        }
+                    });
+                    invalidEntityNodes.forEach((node) => {
+                        if (existingInternetEntityNode) {
+                            // Use the existing INTERNET entity node if it exists
+                            // eslint-disable-next-line no-param-reassign
+                            node.entity = existingInternetEntityNode.entity;
```

Comment:
just curious, why are we overriding here instead of collapsing the two entities into one?

Reply:
I'm not sure I 100% know what you are asking. In a way, we are collapsing the entities into one.

When we get the network response, if there are unknown external connections to/from any deployment in scope, we will get a node with an entity like:
```js
{
    type: 'INTERNET',
    id: 'afa-12345678-1234-1234'
}
```
and if there are _internal_ connections to/from any deployment we'll get a node with an entity like:
```js
{
    type: 5,
    id: 'ada-12345678-1234-5678'
}
```

In each case, there is only ever a single node of type `'INTERNET'` or of type `5` in the response, and each will always have the same id.

The three possible scenarios we run into are:
- The selected deployment scope only has `INTERNET` (external) connections. In this case we do nothing as it is the existing behavior.
- The selected scope only has `5` (internal) connections. In this case we don't know what the id is of the `INTERNET` entity without hard-coding the value, so we just convert the `type` field and keep the `id` field.
- The selected scope has both internal and external connections. In this case, we can't convert the `5` entities, since this would result in two entities of type `INTERNET` with different ids and there will be two nodes of this type displayed in the visualization. Instead, we replace the entire entity for all internal nodes with the known external node entity.

Reply:
wouldn't case 3 potentially cause issues if a deployment is connected to the original external node?

Reply:
I don't think so. From what I understand from the code the connections between nodes aren't by node `id`, but are build based on the index of the node in the response via an `outEdges` property. For example, if the first node in the response was the `INTERNET` node and the second node was the `5 (internal)` node, there could be nodes further down the list with an `outEdge` of `0` to connect it to the `INTERNET` node, or an `outEdge` of `1` to connect it to the `5 (internal)` node.

In case three, when we overwrite the entity property of the second node with the entity from the first node, there will be deployments pointing to both node at index `0` and node at index `1`, which are now the same entity in the graph.

This is at least based on what I've seen in the existing code, but I am relatively new to this area. It holds up correctly when observing the behavior in the graph from my testing.